types/lodash/index.d.ts

```diff
@@ -789,57 +789,57 @@ declare namespace _ {
          * _.differenceWith(objects, [{ 'x': 1, 'y': 2 }], _.isEqual);
          * // => [{ 'x': 2, 'y': 1 }]
          */
-        differenceWith<T>(
-            array: List<T> | null | undefined,
-            values?: List<T>,
-            comparator?: Comparator<T>
-        ): T[];
+        differenceWith<T1, T2>(
```

Comment:
This seems like an odd choice if `T1` and `T2` are not comparable (e.g. when `comparator` is `undefined`). I would recommend you leave the original overload and add this one with both `values` and `comparator` as required, otherwise the entire method is much less type safe.